### PR TITLE
Amend fixtures in tests that don't need model training

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Complete #8124 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/test_model.py [hunk 1]
```diff
@@ -55,13 +55,17 @@
 from tests.core.conftest import DEFAULT_DOMAIN_PATH_WITH_MAPPING, DEFAULT_STACK_CONFIG
 
 
-def test_get_latest_model(trained_rasa_model: str):
-    path_of_latest = os.path.join(os.path.dirname(trained_rasa_model), "latest.tar.gz")
-    shutil.copy(trained_rasa_model, path_of_latest)
+def test_get_latest_model(tmpdir: Path):
```
Comment (wochinge) [inline tests/test_model.py:58]: `tmpdir` is not of type `Path` but of type `LocalPath` which is slightly different. I suggest to use always `tmp_path`
```suggestion
def test_get_latest_model(tmp_path: Path):
```

File: changelog/8124.misc.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Amend fixtures in tests that do not require model training.
```
Comment (wochinge) [inline changelog/8124.misc.md:1]: in my opinion we don't need a changelog as this isn't user facing

**Review by wochinge — APPROVED** [review]
Nicely done! 💯

**Comment by wochinge** [comment]
Re `test_test_core_comparison_after_train`: Would it be possible to use a pre-trained model (e.g. from a fixture) and copy it to `comparison_models/run_1` and so on? It would still train but would use the fixture

File: changelog/8124.misc.md (hunk 2 above)
Comment (alwx) [reply changelog/8124.misc.md:1]: It's a misc entry, so it shouldn't be included to the backlog anyway :)

Comment (wochinge) [reply changelog/8124.misc.md:1]: it actually does https://github.com/RasaHQ/rasa/blob/f52c12a069923ac7cd06fb65bb5b1d8f670fc2e1/CHANGELOG.mdx#L122-L121 . But you're right: doens't really matter 👍🏻

Comment (alwx) [reply changelog/8124.misc.md]: Interesting that the `changelog/README.md` states that it won't be while in fact it's getting included :D need to fix that

File: tests/test_model.py (hunk 1 above)
Comment (ancalita) [reply tests/test_model.py]: That's been amended & also the `test_test_core_comparison_after_train` was changed too 🥳  Let me know your thoughts.

File: tests/cli/test_rasa_test.py [hunk 3]
```diff
@@ -148,42 +150,32 @@ def test_test_core_comparison(
...
-    write_yaml(
-        {"language": "en", "policies": [{"name": "MemoizationPolicy"}]}, "config_1.yml"
-    )
 
-    write_yaml(
-        {"language": "en", "policies": [{"name": "MemoizationPolicy"}]}, "config_2.yml"
-    )
+    path = Path(tmp_path / "comparison_models")
+    path.mkdir()
+    assert path.exists()
```
Comment (wochinge) [inline tests/cli/test_rasa_test.py:160]: I'd remove this line as this only checks if the Python code `path.mkdir` works as expected
```suggestion
```

File: tests/cli/test_rasa_test.py [hunk 4]
```diff
@@ -148,42 +150,32 @@ def test_test_core_comparison(
...
-    )
+    run_one = Path(path / "run_1")
+    run_one.mkdir()
+    assert run_one.exists()
+    shutil.copy(trained_rasa_model, run_one)
+
+    run_two = Path(path / "run_2")
+    run_two.mkdir()
+    assert run_two.exists()
+    shutil.copy(trained_rasa_model, run_two)
```
Comment (wochinge) [inline tests/cli/test_rasa_test.py:170]: optional: some tiny nit picks
```suggestion
    run_one = path / "run_1"
    run_one.mkdir()
    shutil.copy(trained_rasa_model, run_one)

    run_two = path / "run_2"
    run_two.mkdir()
    shutil.copy(trained_rasa_model, run_two)
```

**Review by wochinge — APPROVED** [review]
💯

File: tests/cli/test_rasa_test.py (hunk 3 above)
Comment (ancalita) [reply tests/cli/test_rasa_test.py]: all done now 😃

**Pull request merged**